src/Module/Post/Tag/Remove.php

```diff
@@ -0,0 +1,139 @@
+<?php
+/**
+ * @copyright Copyright (C) 2010-2022, the Friendica project
+ *
+ * @license GNU AGPL version 3 or any later version
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace Friendica\Module\Post\Tag;
+
+use Friendica\App;
+use Friendica\Content\Text\BBCode;
+use Friendica\Core\L10n;
+use Friendica\Core\Renderer;
+use Friendica\Core\Session\Capability\IHandleUserSessions;
+use Friendica\Model\Post;
+use Friendica\Model\Tag;
+use Friendica\Module\Response;
+use Friendica\Util\Profiler;
+use Psr\Log\LoggerInterface;
+
+class Remove extends \Friendica\BaseModule
+{
+	/** @var IHandleUserSessions */
+	private $session;
+
+	public function __construct(IHandleUserSessions $session, L10n $l10n, App\BaseURL $baseUrl, App\Arguments $args, LoggerInterface $logger, Profiler $profiler, Response $response, array $server, array $parameters = [])
+	{
+		parent::__construct($l10n, $baseUrl, $args, $logger, $profiler, $response, $server, $parameters);
+		
+		$this->session = $session;
+	}
+
+	protected function post(array $request = [])
+	{
+		if (!$this->session->getLocalUserId()) {
+			$this->baseUrl->redirect($this->session->get('photo_return'));
+		}
+
+
+		if (isset($request['cancel'])) {
+			$this->baseUrl->redirect($this->session->get('photo_return'));
+		}
+
+		$tags = [];
+		foreach ($request['tag'] ?? [] as $tag => $checked) {
+			if ($checked) {
+				$tags[] = hex2bin(trim($tag));
+			}
+		}
+
+		$this->removeTagsFromItem($this->parameters['item_id'], $tags);
+		$this->baseUrl->redirect($this->session->get('photo_return'));
+	}
+
+	protected function content(array $request = []): string
+	{
+		$photo_return = $this->session->get('photo_return', '');
```

Comment:
Wouldn't a `return` parameter inside the GET parameter be more transparent? I know it from other mechanisms, like OIDC where this is working this way

Reply:
Ah yes, I got carried away trying to fix this stupid feature nobody uses, and I lost sight of this in the process.